docs/src/main/asciidoc/sending-emails.adoc

```diff
@@ -0,0 +1,205 @@
+| bounceAddress | String | the default bounce address
+| host | String | the SMTP server host name
+| port | Integer | the SMTP server port
+| username | String | the username
+| password | String | the password
+| ssl | Boolean | Enables or disables SSL
+| trustAll | Boolean | Set whether to trust all certificates
+| maxPoolSize | Integer | Set the maximum pool size
+| ownHostName | String | The hostname to be used for `HELO/EHLO` and the `Message-ID`
+| keepAlive | Boolean | Set if connection pool is enabled, default is `true`
+| disableEsmtp | Boolean | Set if ESMTP should be tried as first command (EHLO).
+| startTLS | `NONE`, `OPTIONAL`, `REQUIRED` | Set the TLS security mode
+| login | `DISABLED`, `OPTIONAL`, `REQUIRED` | Set the login mode for the connection
+| authMethods | space-separated list | Set the allowed auth methods
+| keyStore | String | Path of the key store
+| keyStorePassword | String | Key store password
+|===
+
+== Sending simple emails
+
+In a JAX-RS resource, or in a bean, you can inject the mailer as follows:
+
+[source, java]
+----
+@Inject
+Mailer mailer;
+
+@Inject
+ReactiveMailer reactiveMailer;
+----
+
+There are 2 APIs:
+
+* `io.quarkus.mailer.Mailer` provides the imperative (blocking and synchronous) API;
+* `io.quarkus.mailer.ReactiveMailer` provides the reactive (non-blocking and asynchronous) API
+
+NOTE: `Mailer` is implemented on top of `ReactiveMailer`
+
+To send a simple email, proceed as follows:
+
+[source, java]
+----
+@GET
+@Path("/simple")
+public Response sendASimpleEmail() {
+    mailer.send(Mail.withText("to@acme.org", "A simple email from quarkus", "This is my body"));
+    return Response.accepted().build();
+}
+
+@GET
+@Path("/async")
+public CompletionStage<Response> sendASimpleEmailAsync() {
+    return reactiveMailer.send(
+            Mail.withText("to@acme.org", "A reactive email from quarkus", "This is my body"))
+            .thenApply(x -> Response.accepted().build());
+}
+----
+
+You can create new `io.quarkus.mailer.Mail` instances from the constructor or from the `Mail.withText` and
+`Mail.withHtml` helper methods. The `Mail` instance lets you add recipients (to, cc, or bcc), set the subject,
+headers, sender (from) address...
+
+You can also send several `Mail` objects in one call:
+
+[source, java]
+----
+mailer.send(mail1, mail2, mail3);
+----
+
+== Sending attachments
+
+To send attachment, just use the `addAttachment` methods on the `io.quarkus.mailer.Mail` instance:
+
+[source,java]
+----
+@GET
+@Path("/attachment")
+public Response sendEmailWithAttachment() {
+    mailer.send(Mail.withText("to@acme.org", "An email from quarkus with attachment",
+            "This is my body")
+            .addAttachment("my-file.txt",
+                "content of my file".getBytes(), "text/plain"));
+    return Response.accepted().build();
+}
+----
+
+Attachments can be created from raw bytes (as shown in the snippet) or files.
+When using a file, the content is read asynchronously.
```

Comment:
This will confuse the blocking API users, I think.

Reply:
Agreed, let me remove this sentence.